pkg/maps/devicemap/devicemap.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package devicemap
+
+import (
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/ebpf"
+	"github.com/cilium/cilium/pkg/mac"
+)
+
+const (
+	MapName    = "cilium_device_map"
+	MaxEntries = 16384
```

Comment:
That's an awfully big number of entries, considering how many network interfaces we'd typically expect to manage :)